lib/Fmt_ast.ml

```diff
@@ -2874,55 +2874,57 @@ and fmt_class_type ?(pro = noop) c ({ast= typ; _} as xtyp) =
   protect c (Cty typ)
   @@
   let {pcty_desc; pcty_loc; pcty_attributes} = typ in
-  update_config_maybe_disabled c pcty_loc pcty_attributes
-  @@ fun c ->
-  let doc, atrs = doc_atrs pcty_attributes in
-  let parens = parenze_cty xtyp in
-  let ctx = Cty typ in
-  let pro ~cmt =
-    pro
-    $ (if cmt then Cmts.fmt_before c pcty_loc else noop)
-    $ fmt_if parens "("
-  and epi ~attrs =
-    fmt_if parens ")"
-    $ (if attrs then fmt_attributes c atrs else noop)
-    $ Cmts.fmt_after c pcty_loc
-    $ fmt_docstring c ~pro:(fmt "@ ") doc
-  in
-  match pcty_desc with
-  | Pcty_constr (name, params) ->
-      let params = List.map params ~f:(fun x -> (x, [])) in
-      hvbox 2
-        ( pro ~cmt:false
-        $ Cmts.fmt_before c pcty_loc
-        $ hovbox 0
-            ( fmt_class_params c ctx params
-            $ fmt_longident_loc c name $ epi ~attrs:true ) )
-  | Pcty_signature {pcsig_self; pcsig_fields} ->
-      let pro = pro ~cmt:true in
-      let epi () = epi ~attrs:true in
-      fmt_class_signature c ~ctx ~pro ~epi pcsig_self pcsig_fields
-  | Pcty_arrow (args, rhs) ->
-      Cmts.relocate c.cmts ~src:pcty_loc
-        ~before:(List.hd_exn args).pap_type.ptyp_loc ~after:rhs.pcty_loc ;
-      let pro =
-        pro ~cmt:true
-        $ fmt_arrow_type c ~ctx ~parens:false ~parent_has_parens:parens args
-            None
-        $ Params.Pcty.arrow c.conf ~rhs
-      in
-      fmt_class_type c ~pro (sub_cty ~ctx rhs) $ epi ~attrs:true
-  | Pcty_extension ext ->
-      hvbox 2 (pro ~cmt:true $ fmt_extension c ctx ext $ epi ~attrs:true)
-  | Pcty_open (popen, cl) ->
-      let pro =
-        pro ~cmt:true
-        $ fmt_open_description c ~keyword:"let open" ~kw_attributes:atrs
-            popen
-        $ str " in"
-        $ Params.Pcty.break_let_open c.conf ~rhs:cl
-      in
-      fmt_class_type c ~pro (sub_cty ~ctx cl) $ epi ~attrs:false
+  let c = update_config c pcty_attributes in
+  (if c.conf.opr_opts.disable.v then pro else noop)
+  $ maybe_disabled c pcty_loc pcty_attributes
```

Comment:
I like the approach but I think it'd be cleaner if it were implemented directly in `update_config_maybe_disabled`. In fact, it ultimately calls `maybe_disabled_k`, which supports this with its last argument.